src/components/workspace-utils.js

```diff
@@ -77,32 +100,104 @@ export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
     loadSnapshotInfo()
   })
 
-  const { name, description, createdDate, source = [] } = snapshotInfo || {}
+
+  // Render
+  const { name: sourceName, description: sourceDescription, createdDate, source = [] } = snapshotInfo || {}
+  const editingDescription = _.isString(newDescription)
 
   return snapshotInfo === undefined ? spinnerOverlay : h(Fragment, [
     div({ style: { padding: '1rem' } }, [
-      div({
-        style: {
-          ...Style.elements.sectionHeader, fontSize: 20,
-          borderBottom: Style.standardLine, paddingBottom: '0.5rem', marginBottom: '1rem'
-        }
-      }, [snapshotName]),
-      h(SnapshotLabeledInfo, { title: 'Source Snapshot Name:', text: name }),
-      h(SnapshotLabeledInfo, { title: 'Creation Date:', text: Utils.makeCompleteDate(createdDate) }),
-      div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, ['Description:']),
-      div([description]),
-      div({ style: Style.dashboard.header }, [`Data Repo Dataset${source.length > 1 ? 's' : ''}`]),
-      _.map(({ dataset: { id, name: datasetName, description: datasetDescription, createdDate: datasetCreatedDate } }) => {
-        return div({
-          key: id,
-          style: { ...Style.elements.card.container, marginBottom: '1rem' }
+      div({ style: Style.elements.card.container }, [
+        div({
+          style: {
+            ...Style.elements.sectionHeader, fontSize: 20,
+            borderBottom: Style.standardLine, paddingBottom: '0.5rem', marginBottom: '1rem'
+          }
         }, [
-          h(SnapshotLabeledInfo, { title: 'Dataset Name:', text: datasetName }),
-          h(SnapshotLabeledInfo, { title: 'Creation Date:', text: Utils.makeCompleteDate(datasetCreatedDate) }),
-          div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, ['Description:']),
-          div([datasetDescription])
-        ])
-      }, source)
+          snapshotName,
+          h(Link, {
+            style: { marginLeft: '0.5rem' },
+            onClick: () => setEditingName(true),
+            'aria-label': 'Edit snapshot name'
+          }, [icon('edit')])
+        ]),
+        div({ style: { ...Style.elements.sectionHeader, marginBottom: '0.2rem' } }, [
+          'Description:',
+          !editingDescription && h(Link, {
+            style: { marginLeft: '0.5rem' },
+            onClick: () => setNewDescription(description),
+            'aria-label': 'Edit description'
+          }, [icon('edit')])
+        ]),
```

Comment:
I see the aria-labels for both name and description  ... could these also have hovertext/tooltips that indicate what they will do?